README.md

```diff
@@ -0,0 +1,177 @@
+# Programming Arcana
+
+The Programming Arcana is an open source programming text book that focuses on teaching introductory programming concepts.
+
+The guiding principles are:
+
+- Transferable concepts are more important than language specifics
+- Build concepts on top of known concepts as much as possible
+- Focus on procedural programming initially, with a view to transitioning to objects
+- Build programs that make it easy to see what is happening in the code
+
+# Code formatter
+
+We use [_clang-format_](https://clang.llvm.org/docs/ClangFormat.html) as our code format for this project.
+
+## Installation
+
+### Linux
+
+_clang-format_ can be installed by:
+
+```bash
+sudo apt install clang-format
+```
+
+### macOS
+
+_clang-format_ can be installed by:
+
+```bash
+brew install clang-format
+```
+
+### Windows
+
+The executable file can be downloaded from [llvm org](https://llvm.org/builds/).
+
+### Other
+
+The _clang-format_ can also be installed with python and nodejs:
+
+```bash
+# python
+python3 -m pip install clang-format
+
+# nodejs
+npm install -g clang-format
+```
+
+The [VS code extension](https://marketplace.visualstudio.com/items?itemName=xaver.clang-format). (optional, clang-format required)
+
+## The format style
+
+The base format style used is [_google_](https://google.github.io/styleguide/cppguide.html), the detailed format options is inside the file [.clang-format](/.clang-format) which is generated with command `clang-format -style=google -dump-config > .clang-format`.
+
+## Usage
+
+To use the code formatter you can use command:
+
+```bash
+clang-format -i <file name>
+```
+
+Or if you have VS code extension installed you can use _right click_ -> _Format Document With ..._ option.
+
+In order to recursively format all files, you can use command:
+
+```bash
+find . -regex '.*\.\(cpp\|c\)' -exec clang-format -style=file -i {} \;
+```
+
+## Check files
```

Comment:
@tancnle Any suggestion for this section title? I do not think this is good.

Reply:
Ideally, it should state what that section is about. I suggest being more specific. Changing _check_ to format or updating formatting or _validate code_

Reply:
Thanks for the suggestion

Reply:
Clang-format is a code formatter, it highlights code style issues. I would vote for `Format code` to keep the language consistent. `Validate` could possibly be inferred as testing as well.